lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -577,15 +577,18 @@ def set_zsort(self, zsort):
 
     def get_vector(self, segments3d):
         """Optimize points for projection."""
-        if len(segments3d):
-            xs, ys, zs = np.row_stack(segments3d).T
-        else:  # row_stack can't stack zero arrays.
-            xs, ys, zs = [], [], []
-        ones = np.ones(len(xs))
-        self._vec = np.array([xs, ys, zs, ones])
-
-        indices = [0, *np.cumsum([len(segment) for segment in segments3d])]
-        self._segslices = [*map(slice, indices[:-1], indices[1:])]
+        # segments3d.shape == (num_faces, num_vertices, 3)
+        if isinstance(segments3d, np.ndarray):
+            assert segments3d.ndim == 3 and segments3d.shape[-1] == 3, \
```

Comment:
I'd go with a more specific error, something like

~~~
raise ValueError(f"segments3d must be a MxNx3 array but got shape {segments3d.shape}"
~~~

Modulo wrapping.